docs/architecture/adr-035-rosetta-api-support.md

```diff
@@ -0,0 +1,238 @@
+
+The constructor follows:
+
+`func New(options Options, network Network) (*Service, error)`
+
+#### Types
+
+`Service` accepts an `Options` `struct` that holds service configuration values, such as the port the service would be listening to:
+
+```golang
+type Options struct {
+    ListenAddress string
+}
+```
+
+The `Network` type holds network-specific properties (i.e. configuration values) and adapters. Pre-configured concrete types will be available for each Cosmos SDK release. Applications can also create their own custom types.
+
+```golang
+type Network struct {
+	Properties rosetta.NetworkProperties
+	Adapter    rosetta.Adapter
+}
+```
+
+A `NetworkProperties` `struct` comprises basic values that are required by a Rosetta API `Service`:
+
+```golang
+type NetworkProperties struct {
+	// Mandatory properties
+	Blockchain          string
+	Network             string
+	SupportedOperations []string
+}
+```
+
+Rosetta API services use `Blockchain` and `Network` as identifiers, e.g. the developers of _gaia_, the application that powers the Cosmos Hub, may want to set those to `Cosmos Hub` and `cosmos-hub-3` respectively.
+
+`SupportedOperations` contains the transaction types that are supported by the library. At the present time, 
+only `cosmos-sdk/MsgSend` is supported in Launchpad. Additional operations will be added in due time.
+
+For Launchpad we will map the amino type name to the operation supported, in Stargate we will use the protoc one.
+
+#### Interfaces
+
+Every SDK version uses a different format to connect (rpc, gRpc, etc), we have abstracted this in what is called the 
+Adapter. This is an interface that defines the methods an adapter implementation must provide in order to be used 
+in the `Network` interface.
+
+Each Cosmos SDK release series will have their own Adapter implementations.
+Developers can implement their own custom adapters as required.
+
+```golang
+type Adapter interface {
+	DataAPI
+	ConstructionAPI
+}
+
+type DataAPI interface {
+	server.NetworkAPIServicer
+	server.AccountAPIServicer
+	server.MempoolAPIServicer
+	server.BlockAPIServicer
+	server.ConstructionAPIServicer
+}
+
+type ConstructionAPI interface {
+	server.ConstructionAPIServicer
+}
+```
+
+### 2. Cosmos SDK Implementation
+
+As described, each Cosmos SDK release series will have version specific implementations of `Network` and `Adapter`, as 
+well as a `NewNetwork` constructor.
+
+Due to separation of interface and implementation, application developers have the option to override as needed, 
+using this code as reference.
+
+```golang
+// NewNetwork returns the default application configuration.
+
+func NewNetwork(options Options) service.Network {
+	cosmosClient := cosmos.NewClient(fmt.Sprintf("http://%s", options.CosmosEndpoint))
+	tendermintClient := tendermint.NewClient(fmt.Sprintf("http://%s", options.TendermintEndpoint))
+
+	return service.Network{
```

Comment:
This diverges from the constructor defined above

Reply:
@jgimeno bump

Reply:
@aaronc True that, yet this is an example, i.e. the actual code would diverge slightly between 0.39 and 0.40 - as described in the comment above: `each Cosmos SDK release series will have version specific implementations of Network and Adapter, as well as a NewNetwork constructor.`

For instance, that's the signature of this very constructor for Stargate:

```
NewNetwork(cdc codec.BinaryMarshaler, options Options) service.Network
```

The constructor's signature for Launchpad follows:

```
NewNetwork(cdc *codec.Codec, options Options) service.Network
```

I hope this helps.